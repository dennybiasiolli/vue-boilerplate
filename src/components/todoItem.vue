<template>
  <v-list-tile @click="()=>{}">
    <v-list-tile-action>
      <v-icon color="indigo">
        keyboard_arrow_right
      </v-icon>
    </v-list-tile-action>
    <v-list-tile-content>
      <v-list-tile-title>{{ todo.text }}</v-list-tile-title>
      <v-list-tile-sub-title>{{ todo.text }}</v-list-tile-sub-title>
    </v-list-tile-content>
  </v-list-tile>
</template>

<script>
export default {
  props: {
    todo: {
      type: Object,
      required: true
    }
  }
}
</script>
