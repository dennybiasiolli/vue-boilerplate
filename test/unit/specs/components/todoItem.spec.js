import { shallow, createLocalVue } from '@vue/test-utils'
import todoItem from '@/components/todoItem'

const localVue = createLocalVue()

describe('todoItem', () => {
  let component, element

  beforeEach(() => {
    component = shallow(todoItem, {
      localVue,
      propsData: { todo: { text: 'ciao' } }
    })
    element = component.element
  })

  test('should match snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(todoItem.props).toEqual({
      todo: {
        type: Object,
        required: true
      }
    })
    expect(todoItem.data).toBeUndefined()
  })

  // Mount an instance and inspect the render output
  test('renders the correct message', () => {
    expect(element.textContent).toContain('ciao')
    expect(component.find('.list__tile__title').element.textContent).toBe('ciao')
    expect(component.find('.list__tile__sub-title').element.textContent).toBe('ciao')
  })
})
