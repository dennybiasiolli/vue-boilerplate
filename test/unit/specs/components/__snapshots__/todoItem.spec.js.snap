// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoItem should match snapshot 1`] = `
<li>
  ciao
</li>
`;
